### PR TITLE
[RFR] Improve autowiring

### PR DESCRIPTION
**Pull request by vincentchalamon**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #1151
| License       | MIT
| Doc PR        | https://github.com/api-platform/docs/pull/214

### CODE REVIEW

File: src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php [hunk 1]
```diff
@@ -13,6 +13,11 @@
 
 namespace ApiPlatform\Core\Bridge\Symfony\Bundle\DependencyInjection;
 
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\QueryCollectionExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\QueryItemExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\FilterInterface;
```
Comment (meyerbaptiste) [inline src/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtension.php]: It should be `ApiPlatform\Core\Api\FilterInterface`.

File: tests/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtensionTest.php [hunk 2]
```diff
@@ -13,14 +13,20 @@
 
 namespace ApiPlatform\Core\Tests\Bridge\Symfony\Bundle\DependencyInjection;
 
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\QueryCollectionExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\QueryItemExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\FilterInterface;
```
Comment (meyerbaptiste) [inline tests/Bridge/Symfony/Bundle/DependencyInjection/ApiPlatformExtensionTest.php]: It should be `ApiPlatform\Core\Api\FilterInterface`.

**Comment by vincentchalamon** [comment]
I'm just thinking about something: filters require `id` option on tag, but I though it would be removed. Is it too soon to auto-configure filters?

**Comment by dunglas** [comment]
It doesn't require anymore because of #862.
However the name of the service must be defined explicilty, I'm not sure if it's compatible or not with the autoconfigure feature.

**Comment by vincentchalamon** [comment]
Maybe autoconfigure is not the right way to declare filters. We should still declare them manually.

BTW @dunglas, #862 requires documentation to be updated.

**Comment by teohhanhui** [comment]
Most of the filters you'd declare in your app are from API Platform anyway, so PSR-4 service discovery and autowiring (there's nothing to be wired up) doesn't really help.

**Comment by dunglas** [comment]
Thank you @vincentchalamon!

**Pull request merged**